eclipse/src/saros/editor/internal/ContributionAnnotationManager.java

```diff
@@ -367,18 +271,17 @@ private void removeAnnotationsForUser(final User user) {
    * Removes all annotations from all annotation models of the given histories. The entries of the
    * histories are removed as well.
    */
-  private void removeFromHistoryAndAnnotationModel(final Collection<History> histories) {
+  private void removeFromHistoryAndAnnotationModel(
+      final Collection<ContributionAnnotationHistory> histories) {
 
     final Set<IAnnotationModel> annotationModels = new HashSet<>();
     final Set<User> users = new HashSet<>();
 
-    for (final History history : histories) {
-      while (!history.entries.isEmpty()) {
-
-        final HistoryEntry entry = history.entries.poll();
-
-        annotationModels.add(entry.annotation.getModel());
-        users.add(entry.annotation.getSource());
+    for (final ContributionAnnotationHistory history : histories) {
+      List<ContributionAnnotation> allAnnotationsInHistory = history.clear();
```

Comment:
Just my 2 cents: Either use the final key constantly or do not use it at all. The code looks pretty weird and you can already see without using GIT Blame that it was written by at least two different developers.